File: MiniDOMFinanziaria/Finanziaria/Prodotti/CategorieProdotto/CategorieProdottoClass.vb
Imports minidom.Databases
Imports minidom.Sistema
Imports minidom.Finanziaria
Imports minidom.Internals

Namespace Internals


    ''' <summary>
    ''' Modulo delle categorie prodotto
    ''' </summary>
    ''' <remarks></remarks>
    <Serializable>
    Public Class CategorieProdottoClass
        Inherits CModulesClass(Of CCategoriaProdotto)

        Public Sub New()
            MyBase.New("modCQSPDCatProd", GetType(CCategorieProdottoCursor), -1)
        End Sub

        Public Function GetItemByName(ByVal nome As String) As CCategoriaProdotto
            nome = Strings.Trim(nome)
            If (nome = "") Then Return Nothing
            For Each c As CCategoriaProdotto In Me.LoadAll
                If (c.Stato = ObjectStatus.OBJECT_VALID AndAlso Strings.Compare(c.Nome, nome) = 0) Then Return c
            Next
            Return Nothing
        End Function


    End Class

End Namespace

Partial Public Class Finanziaria

    Private Shared m_CategorieProdotto As CategorieProdottoClass = Nothing

    Public Shared ReadOnly Property CategorieProdotto As CategorieProdottoClass
        Get
            If (m_CategorieProdotto Is Nothing) Then m_CategorieProdotto = New CategorieProdottoClass
            Return m_CategorieProdotto
        End Get
    End Property

End Class

File: MiniDOMFinanziaria/Finanziaria/Prodotti/CategorieProdotto/CategorieProdottoClass.cs
using System;
using minidom.internals;

namespace minidom
{
    namespace internals
    {


        /// <summary>
    /// Modulo delle categorie prodotto
    /// </summary>
    /// <remarks></remarks>
        [Serializable]
        public class CategorieProdottoClass : CModulesClass<Finanziaria.CCategoriaProdotto>
        {
            public CategorieProdottoClass() : base("modCQSPDCatProd", typeof(Finanziaria.CCategorieProdottoCursor), -1)
            {
            }

            public Finanziaria.CCategoriaProdotto GetItemByName(string nome)
            {
                nome = DMD.Strings.Trim(nome);
                if (string.IsNullOrEmpty(nome))
                    return null;
                foreach (Finanziaria.CCategoriaProdotto c in LoadAll())
                {
                    if (
                        c.Stato == ObjectStatus.OBJECT_VALID 
                        && 
                        DMD.Strings.Compare(c.Nome, nome, true) == 0
                        )
                        return c;
                }

                return null;
            }
        }
    }

    public partial class Finanziaria
    {
        private static CategorieProdottoClass m_CategorieProdotto = null;

        public static CategorieProdottoClass CategorieProdotto
        {
            get
            {
                if (m_CategorieProdotto is null)
                    m_CategorieProdotto = new CategorieProdottoClass();
                return m_CategorieProdotto;
            }
        }
    }
}